biz.aQute.repository/src/aQute/bnd/repository/p2/provider/P2Indexer.java

```diff
@@ -51,6 +57,17 @@
 	private final File						indexFile;
 	private volatile BridgeRepository		bridge;
 	private static final Resource		RECOVERY	= new ResourceBuilder().build();
+	private static final String			P2_CAPABILITY_NAMESPACE	= "bnd.p2";
+	private static final String			MD5_ATTRIBUTE			= "md5";
+	private static final Requirement	MD5_REQUIREMENT;
+	static {
+		try {
+			MD5_REQUIREMENT = new RequirementBuilder(P2_CAPABILITY_NAMESPACE).addAttribute(MD5_ATTRIBUTE, "*")
```

Comment:
Also, I'm curious why this is not:

```java
new RequirementBuilder(P2_CAPABILITY_NAMESPACE).addFilter("(" + MD5_ATTRIBUTE + "=*)").buildSyntheticRequirement();
```

I would assume you need a filter in the requirement to match since attributes on Requirements are generally ignored. Using `addFilter` also avoids the thrown exception so you don't need a try/catch any more.

See https://github.com/bndtools/bnd/blob/master/biz.aQute.bndlib/src/aQute/bnd/osgi/repository/BaseRepository.java#L37-L40